lib/post_destroyer.rb

```diff
@@ -157,10 +157,17 @@ def mark_for_deletion(delete_removed_posts_after = SiteSetting.delete_removed_po
     I18n.with_locale(SiteSetting.default_locale) do
 
       # don't call revise from within transaction, high risk of deadlock
-      @post.revise(@user,
-        { raw: I18n.t('js.post.deleted_by_author', count: delete_removed_posts_after) },
-        force_new_version: true
-      )
+      if @post.is_first_post?
+        @post.revise(@user,
+          { raw: I18n.t('js.topic.deleted_by_author', count: delete_removed_posts_after) },
+          force_new_version: true
+        )
+      else
+        @post.revise(@user,
+          { raw: I18n.t('js.post.deleted_by_author', count: delete_removed_posts_after) },
+          force_new_version: true
+        )
+      end
```

Comment:
One slight nitpick here, the only deference between the lines added here is the key of the translation. I would just go with something like this to remove the deplication.

```
        @post.revise(@user,
          { raw: I18n.t("js.#{@post.is_first_post? ? "topic" : "post"}.deleted_by_author", count: delete_removed_posts_after) },
          force_new_version: true
        )
```

Reply:
I'm not a fan of concatenating translation keys. That makes it so much harder to search for them. I guess the middle ground would be a solution like this. 😉 

```ruby
        raw_key = @post.is_first_post? ? 'js.topic.deleted_by_author' : 'js.post.deleted_by_author'
        @post.revise(@user,
          { raw: I18n.t(raw_key, count: delete_removed_posts_after) },
          force_new_version: true
        )
```

Reply:
This is ok too :)